### ISSUE TITLE
dns not resolving without an internet connection under Yosemite

### ISSUE THREAD

**Issue by joachimroeleveld** [BrianGilbert/OSXAegirInstaller#34]
Since I have a new retina Macbook (late 2013, instead of my old late 2011 model), dnsmasq isn't working anymore when I am not connected to the internet. I saw you already implemented what should be [a solution](http://serverfault.com/questions/22419/set-dns-server-on-os-x-even-when-without-internet-connection), but apparently this doesn't work for me.

The weird thing is, that I installed Aegir on a fresh Mavericks installation (which also ran on my previous Macbook), so I would guess that the problem lies in the hardware configuration; the new macbook has no longer an ethernet connection, only wireless.

The wireless adapter uses 127.0.0.1 as DNS, so that is not the source of the problem.

Any ideas?

**Comment by BrianGilbert** [comment]
@brentratliff you can always add a manual entry to /etc/hosts , that at least stops you from having to install another full stack..  I have a meeting with apple genius next weekend to see if this can get resolved.

**Comment by BrianGilbert** [comment]
you could also try this.. https://github.com/joedj/yosemite_dnsfix  (didn't appear to work for me though)

**Comment by brentratliff** [comment]
I have been adding them in /etc/hosts.  I meant to post that here for others. It wasn't working the first time I added them, so I looked into MAMP, but worked after a reboot.

**Comment by BrianGilbert** [comment]
@brentratliff to flush your local DNS execute the following (so you don't need to reboot when adding new ones)
```
sudo discoveryutil mdnsflushcache;sudo discoveryutil udnsflushcaches;say flushed

**Comment by BrianGilbert** [comment]
This has been escalated to Apple engineers, a workaround that I've confirmed works (but am not electing to use myself) is available at: http://arstechnica.com/apple/2015/01/why-dns-in-os-x-10-10-is-broken-and-what-you-can-do-to-fix-it

**Comment by joachimroeleveld** [comment]
Hmm, I don't think I want to take that risk either. Did you get a response?

**Comment by BrianGilbert** [comment]
Still awaiting a response, you can still use OSX Aegir, you'll just need to add a manual hosts entry for any hosts you want to access while disconnected.

**Comment by joachimroeleveld** [comment]
Yes, I know. I still do all Drupal development on OSX Aegir. Indeed, for every site that I need to access offline, I add a line in the hosts file. Not ideal, but it works.

**Comment by BrianGilbert** [comment]
Response from applecare is that engineers said it's "Expected behaviour" no idea if they will ever fix it.. here is video I made showing the issue (no resolver available when you are disconnected from networks).

https://www.youtube.com/watch?v=3m9OI_AjCx8

https://news.ycombinator.com/item?id=8969581

**Comment by joachimroeleveld** [comment]
Well, that's the second time I experience such behavior from Apple. Half a year ago they screwed me by not providing a solution for my fried [#MacbookPro2011](http://appleinsider.com/articles/14/08/12/apple-ignores-calls-to-fix-2011-macbook-pro-failures-as-problem-grows) and now this. I'm starting to dislike the company more and more.

**Comment by stevestl** [comment]
Good info. Disappointing. 

I could use a tip on modifying my hosts file. I have the following in there and I still can't work in my Apache server offline. What am I missing? Thanks.

127.0.0.1       localhost
255.255.255.255 broadcasthost
::1             localhost
fe80::1%lo0     localhost

127.0.0.1       home.dev
fe80::1%lo0     home.dev

**Comment by dave0783** [comment]
Any update on this?

**Comment by BrianGilbert** [comment]
@dave0783 supposedly you can workaround this using a loopback device, I've just not had enough time to try and figure it out.

**Comment by BrianGilbert** [comment]
Resolved as of 10.10.4 (14E26a), Apple have changed back to mDNSResponder instead of discoveryd!
